src/contracts/pods/EigenPod.sol

```diff
@@ -488,274 +451,182 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-
-        // 1. Balance updates should be more recent than the most recent update
-        require(
-            validatorInfo.mostRecentBalanceUpdateTimestamp < oracleTimestamp,
-            "EigenPod.verifyBalanceUpdate: Validators balance has already been updated for this timestamp"
-        );
-
-        // 2. Balance updates should only be performed on "ACTIVE" validators
-        require(
-            validatorInfo.status == VALIDATOR_STATUS.ACTIVE, 
-            "EigenPod.verifyBalanceUpdate: Validator not active"
-        );
+    function _verifyCheckpointProof(
+        ValidatorInfo memory validatorInfo,
+        uint64 checkpointTimestamp,
+        bytes32 balanceContainerRoot,
+        BeaconChainProofs.BalanceProof calldata proof
+    ) internal returns (int128 balanceDeltaGwei, uint64 exitedBalanceGwei) {
+        uint40 validatorIndex = uint40(validatorInfo.validatorIndex);
+        
+        // Verify validator balance against `balanceContainerRoot`
+        uint64 prevBalanceGwei = validatorInfo.restakedBalanceGwei;
+        uint64 newBalanceGwei = BeaconChainProofs.verifyValidatorBalance({
+            balanceContainerRoot: balanceContainerRoot,
+            validatorIndex: validatorIndex,
+            proof: proof
+        });
+        
+        // Calculate change in the validator's balance since the last proof
+        if (newBalanceGwei != prevBalanceGwei) {
+            balanceDeltaGwei = _calcBalanceDelta({
+                newAmountGwei: newBalanceGwei,
+                previousAmountGwei: prevBalanceGwei
+            });
 
-        // 3. Balance updates should only be made before a validator is fully withdrawn. 
-        // -- A withdrawable validator may not have withdrawn yet, so we require their balance is nonzero
-        // -- A fully withdrawn validator should withdraw via verifyAndProcessWithdrawals
-        if (validatorFields.getWithdrawableEpoch() <= _timestampToEpoch(oracleTimestamp)) {
-            require(
-                validatorEffectiveBalanceGwei > 0,
-                "EigenPod.verifyBalanceUpdate: validator is withdrawable but has not withdrawn"
-            );
+            emit ValidatorBalanceUpdated(validatorIndex, checkpointTimestamp, newBalanceGwei);
         }
 
-        // Verify passed-in validatorFields against verified beaconStateRoot:
-        BeaconChainProofs.verifyValidatorFields({
-            beaconStateRoot: beaconStateRoot,
-            validatorFields: validatorFields,
-            validatorFieldsProof: validatorFieldsProof,
-            validatorIndex: validatorIndex
-        });
+        validatorInfo.restakedBalanceGwei = newBalanceGwei;
+        validatorInfo.lastCheckpointedAt = checkpointTimestamp;
 
-        // Done with proofs! Now update the validator's balance and send to the EigenPodManager if needed
+        // If the validator's new balance is 0, mark them withdrawn
+        if (newBalanceGwei == 0) {
```

Comment:
they may be slashed to 0, in which case if they make subsequent deposits, those deposits are wrecked since the validator is marked as withdrawn

if validators are getting slashed, there are bigger issues to deal with, so seems fine

noting the edge case though

Reply:
> those deposits are wrecked since the validator is marked as withdrawn

nope, those deposits would get picked up in the next checkpoint as `podBalanceGwei` and you'd be able to exit them the same as consensus rewards or other unaccounted-for native ETH

Reply:
ah yep u right